c/src/lib/libbsp/arm/shared/stm32fxxxx/sdram/hal-sdram.c

```diff
@@ -0,0 +1,283 @@
+/**
+ * @file sdram.c
+ *
+ * @ingroup sdram
+ *
+ * @brief A set of utility functions used to configure external SDRAM
+ *
+ */
+
+/*
+ * Copyright (c) 2015 Vecna Technologies, Inc.
+ *
+ * The license and distribution terms for this file may be
+ * found in the file LICENSE in this distribution or at
+ * http://www.rtems.com/license/LICENSE.
+ */
+
+#include <hal-utils.h>
+#include <stm32f-processor-specific.h>
+#include <hal-sdram-interface.h>
+#include <hal-error.h>
+
+#include stm_processor_header(TARGET_STM_PROCESSOR_PREFIX)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, dma)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, sdram)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, cortex)
+#include stm_ll_header(TARGET_STM_PROCESSOR_PREFIX, fmc)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, gpio)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, rcc_ex)
+
+#if defined(EXTERNAL_SDRAM)
+
+static SDRAM_HandleTypeDef hsdram;
+static FMC_SDRAM_TimingTypeDef SDRAM_Timing;
+static FMC_SDRAM_CommandTypeDef command;
+
+/**
+ * @brief  Configure the MPU attributes as Write Through for SRAM1/2.
+ * @note   The Base Address is 0x20010000 since this memory interface is the AXI.
+ *         The Region Size is 256KB, it is related to SRAM1 and SRAM2  memory size.
+ * @param  None
+ * @retval None
+ */
+void MPU_Config(
+  void
+)
+{
+  MPU_Region_InitTypeDef MPU_InitStruct;
+
+  /* Disable the MPU */
+  HAL_MPU_Disable();
+
+  /* Configure the MPU attributes as WT for SRAM */
+  MPU_InitStruct.Enable           = MPU_REGION_ENABLE;
+  MPU_InitStruct.BaseAddress      = 0x20010000;
```

Comment:
Create #defines for the start of internal SRAM and external SDRAM.  Create a way to configure either one using configure.ac values.